MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.App.Tests/Api/Controllers/ListControllerApplicationTests.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using MyPerfectOnboarding.Api.Controllers;
+using MyPerfectOnboarding.Contracts.Database;
+using MyPerfectOnboarding.Contracts.Models;
+using MyPerfectOnboarding.Contracts.Services.Generators;
+using MyPerfectOnboarding.Contracts.Services.Location;
+using MyPerfectOnboarding.Dependency;
+using MyPerfectOnboarding.Dependency.Containers;
+using MyPerfectOnboarding.Tests.Utils.Builders;
+using MyPerfectOnboarding.Tests.Utils.Comparers;
+using MyPerfectOnboarding.Tests.Utils.Extensions;
+using NSubstitute;
+using NUnit.Framework;
+using Unity;
+
+namespace MyPerfectOnboarding.App.Tests.Api.Controllers
+{
+    [TestFixture]
+    internal class ListControllerApplicationTests
+    {
+        private ListController _listController;
+        private IUrlLocator _location;
+        private ITimeGenerator _timeGenerator;
+        private IGuidGenerator _guidGenerator;
+        private IListRepository _listRepository;
+
+        ListItem[] _items =
+        {
+            ListItemBuilder.CreateItem("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC", "aaaaa", "1589-12-03", "1896-04-07"),
+            ListItemBuilder.CreateItem("11AC59B7-9517-4EDD-9DDD-EB418A7C1644", "dfads", "4568-06-23", "8569-08-24")
+        };
+
+        [SetUp]
+        public void Init()
+        {
+
+
+            var unityContainer = new UnityContainer();
+            var container = new Container(unityContainer);
+
+            RegisterTypes(container);
+
+            _listController = container.UnityContainer.Resolve<ListController>();
```

Comment:
I don't like this construct. It is nothing like the way you use it in the application itself. Let's extend the `CreateContainer` (or `MockContainer` (or `…`)) method and return `IDependencyResolver` from it (along with `IContainer`). If changing the type raises your eyebrow, feel free to introduce a simple extension method, however, it is only used at one place and "hidden" in a  `CreateController` (or `MockController` (or `…`)) method 🙃